iOS/Flows/Conversation/Cells/TimeMachineCollectionViewLayout.swift

```diff
@@ -297,47 +251,41 @@ class TimeMachineCollectionViewLayout: UICollectionViewLayout {
 
     // MARK: - Update Animation Handling
 
-    /// Ids of items that are being inserted.
-    private var insertedIds: Set<String> = []
-    /// Ids of items that are being deleted.
-    private var deletedIds: Set<String> = []
-    /// Ids of items that that were visible before the animation started.
-    private var idsVisibleBeforeAnimation: Set<String> = []
-    /// How much to adjust the proposed scroll offset.
-    private var scrollOffsetAdjustment: CGFloat = 0
+    /// Index paths of items that are being inserted.
+    private var insertedIndexPaths: Set<IndexPath> = []
     /// The z position before update animations started
     private var zPositionBeforeAnimation: CGFloat = 0
+    /// If true, we should adjust the scroll offset so the previously focused item remains in focus.
```

Comment:
Everything from here to the end is about keep our place when deleting/inserting messages. No need to understand it all, just know that it's where you should look if things aren't working quite right.